VideoLocker/src/main/java/org/edx/mobile/authentication/LoginTask.java

```diff
@@ -48,9 +50,33 @@ public AuthResponse call() throws Exception {
 
         // get profile of this user
         if (response.isSuccess()) {
-            response.profile = api.getProfile();
+            try {
+                response.profile = api.getProfile();
+            } catch (IOException e) {
+                throw new RetroHttpException(RetrofitError.networkError(null, e));
+            } catch (Exception e) {
+                /* TODO: Move the API for getting the user profile data
+                 * to Retrofit, and then post a message appropriate for
+                 * the type of error encountered during the request.
+                 */
+                throwLoginException();
+            }
+            if (!response.hasValidProfile()) {
+                throwLoginException();
+            }
 
         }
         return response;
     }
+
+    /**
+     * Throw a generic exception about login failure.
+     *
+     * @throws LoginException The exception.
```

Comment:
All parameters and checked exception throwing declarations need to be documented in the javadoc. `LoginException` is already declared as the type of the exception, so adding it in the description would be redundant, and since the method itself is about throwing the exception, it seems fine to have the description of the exception stubbed in favor of it's description in the method's description itself.